plugins/generator-1.15.2/forge-1.15.2/templates/mob.java.ftl

```diff
@@ -438,6 +440,26 @@ import net.minecraft.block.material.Material;
 				if (source == DamageSource.LIGHTNING_BOLT)
 					return false;
 			</#if>
+			<#if data.immuneToExplosion>
+				if (source == DamageSource.causeExplosionDamage(this))

```

Comment:
Did you test this, because I am not very certain == will work with non-static fields, == compares pointers to the objects and they can't really match in this case. Same for other cases like this one (trident).

Reply:
There is this above : `if (source == DamageSource.LIGHTNING_BOLT)` so it must work, mustn't it?

Reply:
> so it must work, mustn't it?

> I am not very certain == will work with non-static fields

DamageSource.LIGHTNING_BOLT is a static object, also with a constant object pointer, so you can use pointer comparison ==. 

Please test in the game, this will not work. Side note, you should test this anyways, I am not sure why you still don't test your PRs.